install.sh

```diff
@@ -89,6 +89,29 @@ if [ -n "$pkg" ] && [ "$pkg" != "server" -a "$pkg" != "conf" -a "$pkg" != "kerne
     exit 1
 fi
 
+patchelf() {
+    # patchelf comes from the build system, so it needs the build system's ld.so and
+    # shared libraries. We can't use patchelf on patchelf itself, so invoke it via
+    # ld.so.
+    LD_LIBRARY_PATH="$PWD/libreloc" libreloc/ld.so libexec/patchelf "$@"
+}
+
+adjust_bin() {
+    local bin="$1"
+    # We could add --set-rpath too, but then debugedit (called by rpmbuild) barfs
+    # on the result. So use LD_LIBRARY_PATH in the thunk, below.
+    patchelf \
+	--set-interpreter "$prefix/libreloc/ld.so" \
+	"$root/$prefix/libexec/$bin"
+    cat > "$root/$prefix/bin/$bin" <<EOF
+#!/bin/bash -e
+export GNUTLS_SYSTEM_PRIORITY_FILE="\${GNUTLS_SYSTEM_PRIORITY_FILE-$prefix/libreloc/gnutls.config}"
+export LD_LIBRARY_PATH="$prefix/libreloc"
+exec -a "\$0" "$prefix/libexec/$bin" "\$@"
```

Comment:
Since we are creating a script anyway, it seems better to use a CWD relative path for the interpreter. That way the script can be similar to

cd $prefix/libexec
exec -a "\$0" ./$bin "\$@"

And we don't need to patch the binary.

Reply:
We can't change CWD from under the executable's feet.

Reply:
Fair point. LGTM